lib/Chat/MessageParser.php

```diff
@@ -103,4 +107,30 @@ protected function setActor(Message $message): void {
 			$displayName
 		);
 	}
+
+	public function isSharedFile(string $message): bool {
+		$parameters = $this->getParametersFromMessage($message);
+		return !empty($parameters['share']);
+	}
+
+	private function getParametersFromMessage(string $message): array {
+		$data = json_decode($message, true);
+		if (!\is_array($data)) {
+			return [];
+		}
+		return $data['parameters'];
+	}
+
+	public function fileOfMessageExists(string $message): bool {
+		$parameters = $this->getParametersFromMessage($message);
+		if (empty($parameters['share'])) {
+			return false;
+		}
+		try {
+			$this->shareProvider->getShareById($parameters['share']);
```

Comment:
This is not scaling.

Please call this method in the controllers entry points before the loops:
```php
$this->preloadShares($comments);
```
Similar to this:
https://github.com/nextcloud/spreed/blob/8a4bd006212beafd6e1c5593b46a7bffd0bdcf17/lib/Controller/ChatController.php#L466